### DISCUSSION TITLE
Add additional certificates to LB listener

### DISCUSSION THREAD

**Discussion by fsanzdev** [gruntwork-io/knowledge-base#476] [Help] [answered]
Hi, 

How can I add additional ACM certs to a listener in a load balancer using catalog service? I can't see any option to do that.

Thanks

---

<ins datetime="2022-06-16T18:29:07Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108794">Tracked in ticket #108794</a></p>
</ins>

**Comment by yorinasub17** [comment]
You can only specify a single ACM Cert per listener. This is an AWS limitation.

If your use case is to serve multiple different domains from the load balancer, the standard way to address this is to add additional domains (via SAN directives) to the ACM cert so that they match against all the domains you need.

**Reply by fsanzdev** [reply]
Thanks Yori, but I think it's quite a while since this is possible https://aws.amazon.com/blogs/aws/new-application-load-balancer-sni/ and it's a significant limitation not to have it available in the modules since everytime you need to add a new domain you have to re-create the certificate, remove the listener, regenerate the listener and redeploy the services linked with that listener. 

Having the option to add additional certs to the listener would fix this issue.

**Reply by yorinasub17** [reply]
Ah apologies for that, you are correct. This is supported, just in a different resource. I completely misread the limitation in the terraform docs. I have added this as a feature request on `terraform-aws-service-catalog` so that it goes in our work queue: https://github.com/gruntwork-io/terraform-aws-service-catalog/issues/1546

Note that we are currently buried with other initiatives, so it may be a while before we can get through to this. In the meantime, your best options for workarounds are:

- Create [a wrapper module to extend the service catalog module](https://docs.gruntwork.io/reference/services/intro/create-your-own-service-catalog) with that new resource mentioned in the ticket. (Cleanest, but may make migration in the future harder if we add support).
- Inject the resource into the service catalog using Terragrunt `generate` blocks. (Dirty, but makes migration easier in the future due to `state mv` calls).
- Fork our module. (Adds maintenance overhead, but makes migration easier in the future)
- Submit a PR adding support for the resource (Least maintenance overhead)

**Comment by yorinasub17** [comment, marked as answer]
This is now supported starting with version https://github.com/gruntwork-io/terraform-aws-service-catalog/releases/tag/v0.92.0 of the service catalog.
